### PR TITLE
checking if divergence of B is 0 at initialization

### PR DESCRIPTION
**Pull request by guadabsb15**
## PR summary
Adding a check during the initialization, for the divergence of magnetic field to be 0 

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
I don't think you need ghost cells, I think you should create a cell-centered div B just like we do for the derived variables, e.g.,
```
divB(i,,j,k) = (Bx(i+1,j,k) - Bx(i,j,k))/dx + (By(i,j+1,k) - By(i,j,k))/dy + (Bz(i,j,k+1) - Bz(i,j,k))/dz
```

also, I think you need to normalize the check by the magnitude of B, since roundoff will be relatively to the size of the field.

**Comment by guadabsb15** [comment]
> I don't think you need ghost cells, I think you should create a cell-centered div B just like we do for the derived variables, e.g.,
> 
> ```
> divB(i,,j,k) = (Bx(i+1,j,k) - Bx(i,j,k))/dx + (By(i,j+1,k) - By(i,j,k))/dy + (Bz(i,j,k+1) - Bz(i,j,k))/dz
> ```
> 
> also, I think you need to normalize the check by the magnitude of B, since roundoff will be relatively to the size of the field.

I think I am doing that now

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -3485,6 +3489,67 @@ Castro::add_magnetic_e( MultiFab& Bx,
...
+      {
+          
+          Real divB = (Bx_arr(i+1,j,k) - Bx_arr(i,j,k))/dx[0] +
+                      (By_arr(i,j+1,k) - By_arr(i,j,k))/dx[1] + 
+                      (Bz_arr(i,j,k+1) - Bz_arr(i,j,k))/dx[2];
+
+          Real magB = std::sqrt(Bx_arr(i,j,k) * Bx_arr(i,j,k) + 
+                                By_arr(i,j,k) * By_arr(i,j,k) +
+                                Bz_arr(i,j,k) * Bz_arr(i,j,k));
+
```
Comment (zingale) [inline Source/driver/Castro.cpp:3531]: I think we want magB to be cell-centered, so you should average the face centered B to centers and then compute mag B of that

File: Source/driver/Castro.cpp [hunk 2]
```diff
@@ -3485,6 +3489,67 @@ Castro::add_magnetic_e( MultiFab& Bx,
...
+          if (std::abs(divB) > 1.0e-10*magB){
+             fail_divB = 1; 
+          }
+          
+
+          return {fail_divB}; 
+      });
+
+      ReduceTuple hv = reduce_data.value();
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:3543]: This should go outside the MFIter loop

File: Source/driver/Castro.cpp (hunk 1 above)
Comment (guadabsb15) [reply Source/driver/Castro.cpp:3548]: true, I just changed it

File: Source/driver/Castro.cpp (hunk 2 above)
Comment (guadabsb15) [reply Source/driver/Castro.cpp]: thanks for pointing that out

**Pull request merged**